### PR TITLE
[Android] Enable ARM 64-bit support for Crosswalk

### PR DESCRIPTION
**Pull request by shawngao5**
BUG=XWALK-2839

### CODE REVIEW

File: app/tools/android/make_apk.py [hunk 1]
```diff
@@ -494,14 +507,12 @@ def MakeApk(options, app_info, manifest):
     else:
       # If the arch option is unspecified, all of available platform APKs
       # will be generated.
-      valid_archs = ['x86', 'x86_64', 'armeabi-v7a']
+      valid_archs = ['x86', 'x86_64', 'armeabi-v7a', 'arm64-v8a']
+      print valid_archs 
+      print available_archs
```
Comment (halton) [inline app/tools/android/make_apk.py]: remove this debugging calls

File: app/tools/android/make_apk_test.py [hunk 2]
```diff
@@ -180,6 +182,8 @@ def archs():
                                        'x86_64', 'libxwalkcore.so')
     arm_native_lib_path = os.path.join('xwalk_core_library', 'libs',
                                        'armeabi-v7a', 'libxwalkcore.so')
+    arm64_native_lib_path = os.path.join('xwalk_core_library', 'libs',
+                                       'arm64-v8a', 'libxwalkcore.so')
```
Comment (halton) [inline app/tools/android/make_apk_test.py]: align with `'`

File: app/tools/android/make_apk_test.py [hunk 3]
```diff
@@ -906,6 +927,21 @@ def testArch(self):
         self.assertFalse(os.path.isfile('Example_1.0.0._arm.apk'))
       self.assertFalse(os.path.isfile('Example_1.0.0_x86.apk'))
       self.assertFalse(os.path.isfile('Example_1.0.0_x86_64.apk'))
+      self.assertFalse(os.path.isfile('Example_1.0.0_arm64.apk'))
+      Clean('Example', '1.0.0')
+
+      cmd = ['python', 'make_apk.py', '--name=Example', '--app-version=1.0.0',
+             '--package=org.xwalk.example', '--app-url=http://www.intel.com',
```
Comment (halton) [inline app/tools/android/make_apk_test.py]: how about `http://www.crosswalk-project.org` instead, Crosswalk is open source project

Comment (halton) [inline app/tools/android/make_apk_test.py]: And seems, you missing `--arch=arm64`

File: app/tools/android/make_apk.py [hunk 4]
```diff
@@ -494,14 +507,12 @@ def MakeApk(options, app_info, manifest):
     else:
       # If the arch option is unspecified, all of available platform APKs
       # will be generated.
-      valid_archs = ['x86', 'x86_64', 'armeabi-v7a']
+      valid_archs = ['x86', 'x86_64', 'armeabi-v7a', 'arm64-v8a']
```
Comment (rakuco) [inline app/tools/android/make_apk.py]: Why not use the arch folder names and avoid the need for `ConvertArchFolderToArchName()` to exist altogether?

```python
for arch in AllArchitectures():
  if ConvertArchNameToFolder(arch) in available_archs:
    # yadda, yadda
  else:
    # yadda, yadda
```

File: app/tools/android/make_apk.py [hunk 5]
```diff
@@ -549,7 +560,7 @@ def main(argv):
   parser.add_option('--mode', choices=('embedded', 'shared'),
                     default='embedded', help=info)
   info = ('The target architecture of the embedded runtime. Supported values '
```
Comment (rakuco) [inline app/tools/android/make_apk.py]: You could use `AllArchitectures()` here too so you don't need to hardcode the list.

```python
info = ('The target architecture of the embedded runtime. Supported values: %s. If not specified, APKs for all available architectures will be generated.' % ', '.join(AllArchitectures()))
```

File: app/tools/android/make_apk.py [hunk 6]
```diff
@@ -494,19 +497,25 @@ def MakeApk(options, app_info, manifest):
...
       # will be generated.
-      valid_archs = ['x86', 'x86_64', 'armeabi-v7a']
-      for arch in valid_archs:
-        if arch in available_archs:
-          if arch.find('arm') != -1:
-            options.arch = 'arm'
-          else:
-            options.arch = arch
-          print("options.arch:", options.arch)
+      valid_archs = ['x86', 'x86_64', 'armeabi-v7a', 'arm64-v8a']
```
Comment (rakuco) [inline app/tools/android/make_apk.py]: You still need to remove these debugging bits.

File: app/tools/android/make_apk.py [hunk 7]
```diff
@@ -494,19 +497,25 @@ def MakeApk(options, app_info, manifest):
...
+      print available_archs
+      for arch in AllArchitectures():
+        if ConvertArchNameToArchFolder(arch) in available_archs:
+          options.arch = arch
           Execution(options, app_info)
           packaged_archs.append(options.arch)
         else:
           print('Warning: failed to create package for arch "%s" '
                 'due to missing native library' % arch)
+#      for arch in valid_archs:
```
Comment (rakuco) [inline app/tools/android/make_apk.py]: Same here.

File: app/tools/android/make_apk.py [hunk 8]
```diff
@@ -548,9 +557,12 @@ def main(argv):
           'Set the default mode as \'embedded\'. For example: --mode=embedded')
   parser.add_option('--mode', choices=('embedded', 'shared'),
                     default='embedded', help=info)
-  info = ('The target architecture of the embedded runtime. Supported values '
-          'are \'x86\' \'x86_64\' and \'arm\'. Note, if undefined, APKs for '
-          'all possible architestures will be generated.')
+  info = ('The target architecture of the embedded runtime. Supported values: '
+          '%s. If not specified, APKs for all available architectures will be '
+          'generated.' % ', '.join(AllArchitectures()))
+#  info = ('The target architecture of the embedded runtime. Supported values '
```
Comment (rakuco) [inline app/tools/android/make_apk.py]: And here.

File: app/tools/android/make_apk_test.py [hunk 9]
```diff
@@ -370,7 +389,7 @@ def testAppBigVersionCodeBase(self):
            '--description=a sample application',
            '--app-versionCodeBase=3000000',
            arch,
-           '--app-url=http://www.intel.com',
+           '--app-url=http://www.crosswalk-project.org',
```
Comment (rakuco) [inline app/tools/android/make_apk_test.py]: I think Halton meant you should do this for the new code you are adding. Changing all the existing occurrences is unrelated to what your patch intends to do, and only makes it bigger; I support changing these existing URLs, but separately.

Comment (halton) [inline app/tools/android/make_apk_test.py]: Yes, please do it separately for changing the existing code.

**Comment by rakuco** [comment]
lgtm, thanks!

**Comment by halton** [comment]
LGTM.

**Pull request merged**